src/ChainedMap.js

```diff
@@ -53,8 +74,23 @@ module.exports = class extends Chainable {
     return this;
   }
 
-  merge(obj) {
-    Object.keys(obj).forEach(key => this.set(key, obj[key]));
+  merge(obj, omit = []) {
```

Comment:
This and the other merge/omit parts could be a separate commit in the PR from the before/after parts :-)